src/instruments/src/SD/Pages/Hyd/Hyd.tsx

```diff
@@ -209,243 +123,329 @@ const RAT = ({ x, y }: RATProps) => {
+
+            <text className={lowAirPress ? 'Large Amber' : 'Hide'} x={20} y={-80}>LO AIR</text>
+            <text className={lowAirPress ? 'Large Amber' : 'Hide'} x={20} y={-50}>PRESS</text>
+        </SvgGroup>
+    );
+};
+
+type YellowElecPumpProps = {
+    pumpPushbuttonOn: boolean,
+    pressure: number,
+    enginePumpPressureLowSwitch: boolean,
+};
+
+const YellowElecPump = ({ pumpPushbuttonOn, pressure, enginePumpPressureLowSwitch }: YellowElecPumpProps) => {
+    const [ACBus2IsPowered] = useSimVar('L:A32NX_ELEC_AC_2_BUS_IS_POWERED', 'bool', 1000);
+
+    let elecHorizontalLineFormat: string;
+    let verticalLineFormat: string;
+    let elecTriangleFill: number;
+    let elecTriangleColour: string;
+
+    if (!pumpPushbuttonOn) {
+        elecTriangleFill = 0;
+        elecHorizontalLineFormat = 'Hide';
+        elecTriangleColour = 'White';
+    } else {
+        elecTriangleFill = 1;
+        elecHorizontalLineFormat = pressure <= 1450 ? 'AmberLine' : 'GreenLine';
+        elecTriangleColour = pressure <= 1450 ? 'Amber' : 'Green';
+    }
+
+    if ((enginePumpPressureLowSwitch && !pumpPushbuttonOn) || pressure <= 1450) {
+        verticalLineFormat = 'AmberLine';
+    } else {
+        verticalLineFormat = 'GreenLine';
+    }
+
+    return (
+        <>
+            <text
+                id="ELEC-right"
+                className={!ACBus2IsPowered ? 'Large Amber' : 'Large'}
+                x={676}
+                y={292}
+            >
+                ELEC
+            </text>
+            <Triangle x={642} y={283} scale={4 / 3} colour={elecTriangleColour} fill={elecTriangleFill} orientation={-90} />
+            <line className={elecHorizontalLineFormat} x1={631} y1={283} x2={642} y2={283} />
+            <line className={verticalLineFormat} x1={630} y1={217} x2={630} y2={283} />
         </>
     );
 };
 
+enum TransferColor {
+    Green = 'Green',
+    Amber = 'Amber',
+}
+
+enum TransferState {
+    GreenToYellow,
+    YellowToGreen,
+    None
+}
+
 type PTUProps = {
     x: number,
     y: number,
-    ptuScenario: string
+    yellowPressure: number,
+    greenPressure: number,
+    yellowPumpLowPressure: boolean,
+    greenPumpLowPressure: boolean,
+    yellowQuantity: number,
+    greenQuantity: number,
+    ptuControlValveOff: boolean
+    yellowElecPumpOn: boolean,
 }
 
-const PTU = ({ x, y, ptuScenario } : PTUProps) => {
-    const semiCircleD = `M${x - 16},${y} C${x - 16},${y + 24} ${x + 16},${y + 24} ${x + 16},${y}`;
+const shouldTransferActivate = (
+    lowerPressureSystemQuantity: number,
+    lowerPressureSystemPressure: number,
+    highPressureSystemPressure: number,
+    lowerPressureSystemPumpLowPress: boolean,
+    yellowElecPumpOn: boolean,
+    transferDirection: TransferState,
+) => {
+    if (transferDirection === TransferState.None) {
+        return false;
+    }
+
+    const litersPerGallon = 3.79;
```

Comment:
Maybe we can make this a constant in this file? I see 3.79 appearing as a magic number later.